gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -277,19 +275,19 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                         it.safeAs<Task>()?.name == kspTaskName
                 }
                 kspTask.dependsOn(deps)
-                kspTask.source(kotlinCompileTask.source)
-                if (kotlinCompileTask is AbstractKotlinCompile<*>) {
-                    val sourceRoots = kotlinCompileTask.getSourceRoots()
-                    if (sourceRoots is SourceRoots.ForJvm) {
-                        kspTask.source(sourceRoots.javaSourceRoots)
-                    }
+                kspTask.setSource(kotlinCompileTask.sources)
+                if (kotlinCompileTask is KotlinCompile) {
+                    kspTask.setSource(kotlinCompileTask.javaSources)
                 }
             } else {
-                kotlinCompilation.allKotlinSourceSets.forEach { sourceSet -> kspTask.source(sourceSet.kotlin) }
+                kotlinCompilation.allKotlinSourceSets.forEach { sourceSet ->
+                    kspTask.setSource(sourceSet.kotlin)
+                }
                 if (kotlinCompilation is KotlinCommonCompilation) {
-                    kspTask.source(kotlinCompilation.defaultSourceSet.kotlin)
+                    kspTask.setSource(kotlinCompilation.defaultSourceSet.kotlin)
                 }
             }
+            kspTask.exclude { kspOutputDir.isParentOf(it.file) }
```

Comment:
What's the use of this?

Reply:
Good catch. It's a leftover from rebase and is removed.